### PR TITLE
Optimize server startup

### PR DESCRIPTION
**Pull request by DamianX**
I tested this three times each, measured from "global vars are initialized" to "reached the end of world/New".
Before:
22.4s
19.2s
19.6s

After:
11s
11.1s
10.1s

Some of the time that was spent here has been entirely shaved off, but most of it been moved so that it happens during the Objects subsystem init, which means that the loop can `CHECK_TICK`, which means that the server isn't completely frozen while that happens, which means that now clients will be able to connect approximately 10 seconds earlier than before.

It's easier to review the changes by looking at the individual commits.

### CODE REVIEW

File: code/game/machinery/computer/HolodeckControl.dm [hunk 1]
```diff
@@ -407,10 +407,11 @@
 /turf/simulated/floor/holofloor/grass
 	name = "lush Grass"
 	icon_state = "grass1"
+
+/turf/simulated/floor/holofloor/grass/create_floor_tile()
 	floor_tile = new/obj/item/stack/tile/grass
 
 /turf/simulated/floor/holofloor/grass/New()
-	floor_tile.New() //I guess New() isn't run on objects spawned without the definition of a turf to house them, ah well.
```
Comment (Exxion) [inline code/game/machinery/computer/HolodeckControl.dm:413]: What did he mean by this?

Comment (DamianX) [reply code/game/machinery/computer/HolodeckControl.dm:413]: Might have been a holdover from the dark ages during which `getFromPool` didn't call New

Comment (Exxion) [reply code/game/machinery/computer/HolodeckControl.dm:413]: How could `getFromPool()` have been called before this line executes though?

Comment (DamianX) [reply code/game/machinery/computer/HolodeckControl.dm:413]: we just don't know

Comment (Exxion) [reply code/game/machinery/computer/HolodeckControl.dm:413]: I actually just checked blame and this particular line *predates pooling*

**Pull request merged**